packages/contracts-bedrock/test/L2/SuperchainERC20.t.sol

```diff
@@ -0,0 +1,292 @@
+        vm.expectRevert("ERC20: burn from the zero address");
+
+        // Call the `burn` function with the zero address
+        vm.prank(BRIDGE);
+        superchainERC20.burn({ _from: ZERO_ADDRESS, _amount: _amount });
+    }
+
+    /// @dev Tests the `burn` bruns the amount and emits the `Burn` event.
+    function testFuzz_burn_succeeds(address _from, uint256 _amount) public {
+        // Ensure `_from` is not the zero address
+        vm.assume(_from != ZERO_ADDRESS);
+
+        // Mint some tokens to `_from` so then they can be burned
+        vm.prank(BRIDGE);
+        superchainERC20.mint(_from, _amount);
+
+        // Get the total supply and balance of `_from` before the burn to compare later on the assertions
+        uint256 _totalSupplyBefore = superchainERC20.totalSupply();
+        uint256 _fromBalanceBefore = superchainERC20.balanceOf(_from);
+
+        // Look for the emit of the `Transfer` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit IERC20.Transfer(_from, ZERO_ADDRESS, _amount);
+
+        // Look for the emit of the `Burn` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit SuperchainERC20.Burn(_from, _amount);
+
+        // Call the `burn` function with the bridge caller
+        vm.prank(BRIDGE);
+        superchainERC20.burn(_from, _amount);
+
+        // Check the total supply and balance of `_from` after the burn were updated correctly
+        assertEq(superchainERC20.totalSupply(), _totalSupplyBefore - _amount);
+        assertEq(superchainERC20.balanceOf(_from), _fromBalanceBefore - _amount);
+    }
+
+    /// @dev Tests the `sendERC20` function burns the sender tokens, sends the message, and emits the `SentERC20` event.
+    function testFuzz_sendERC20_succeeds(address _sender, address _to, uint256 _amount, uint256 _chainId) external {
+        // Ensure `_sender` is not the zero address
+        vm.assume(_sender != ZERO_ADDRESS);
+
+        // Mint some tokens to the sender so then they can be sent
+        vm.prank(BRIDGE);
+        superchainERC20.mint(_sender, _amount);
+
+        // Get the total supply and balance of `_sender` before the send to compare later on the assertions
+        uint256 _totalSupplyBefore = superchainERC20.totalSupply();
+        uint256 _senderBalanceBefore = superchainERC20.balanceOf(_sender);
+
+        // Look for the emit of the `Transfer` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit IERC20.Transfer(_sender, ZERO_ADDRESS, _amount);
+
+        // Look for the emit of the `SentERC20` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit SuperchainERC20.SentERC20(_sender, _to, _amount, _chainId);
+
+        // Mock the call over the `sendMessage` function and expect it to be called properly
+        _mockAndExpect(
+            MESSENGER,
+            abi.encodeWithSelector(
```

Comment:
hehe i meant using `abi.encodeCall` instead of `abi.encodeWithSelector` not moving the `message` declaration within the body, but it's ok

Reply:
ah jaja, will undo that and use `abi.encodeCall` then